host/host.go

```diff
@@ -61,6 +61,9 @@ type Host interface {
 	// A string representation of the host which uniquely identifies it, eg, its FQDN.
 	String() string
 
+	// String representation for the type of connection is used. eg: ssh, localhost, docker
+	Type() string
```

Comment:
I wonder if we'd be better off using good names for the `Host` types, and then:

```go
func GetHostType(hst Host) string {
  return strings.ToLower(reflect.TypeOf(hst).Elem().Type().Name)
}
```

Reply:
Oh, this may be an issue with `h.Host.Type()` below...

Reply:
That seems like a more complex approach, and the use of `reflect` might be a overkill, at least for now.
We can definitely implement it, but I'd like to mature this idea a bit more.